components/api/src/manager.rs

```diff
@@ -121,12 +128,39 @@ async fn with_retries<T>(
     Err(io::Error::new(io::ErrorKind::Other, msg))
 }
 
+#[derive(Debug)]
+struct ServerHandle {
+    inner: Server,
+    #[cfg(windows)]
+    raw_socket: RawSocket,
+}
+
+impl ServerHandle {
+    #[cfg(not(windows))]
+    async fn stop(self) {
+        self.inner.stop(false).await;
+    }
+
+    #[cfg(windows)]
+    #[allow(unsafe_code)]
+    async fn stop(self) {
+        self.inner.stop(false).await;
+
+        // SAFETY: The safety of `from_raw_socket` relies on the fact that the socket is not used
```

Comment:
You may refer this issue https://github.com/actix/actix-web/issues/1249 in this comment to provide better explanation what it is actually happening.